sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala

```diff
@@ -233,7 +233,7 @@ case class GetArrayStructFields(
 case class GetArrayItem(
     child: Expression,
     ordinal: Expression,
-    failOnError: Boolean = SQLConf.get.ansiFailOnElementNotExists)
+    failOnError: Boolean = SQLConf.get.failOnNonexistentElement)
```

Comment:
how about simply using the config key directly? `SQLConf.get.getConf(SQLConf. ANSI_STRICT_INDEX_OPERATOR)`

Reply:
I have updated the PR.